integration-test/src/main.rs

```diff
@@ -0,0 +1,243 @@
+use mongodb::{
+    bson::{doc, Bson},
+    sync::Client,
+    IndexModel,
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::BTreeSet, env, fs, io};
+use thiserror::Error;
+
+const TEST_DATA_DIRECTORY: &str = "resources/integration_test/testdata";
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+struct TestData {
+    dataset: Vec<TestDataEntry>,
+
+    #[serde(skip)]
+    file: String,
```

Comment:
What are we using `file` here for?

Reply:
Exclusively for the error case in `set_test_data_schemas` where I get the datasource (either the view or collection name). _Technically_ since I have validation at file-reading time that exactly one of those appears, it should really be a `panic` in `set_test_data_schemas` since that case can only be hit if there is a bug in the previous code. Maybe I'll update it to panic and drop the `file` field.

Reply:
I decided to keep this since the distinction between an error and a panic in this code is indistinguishable for us since we are the only ones who run it. Keeping the file name is also still useful since it helps indicate which file has the problem in it! lmk if you strongly prefer panic over error or removing the file name.

Reply:
Thanks for clarifying what it's used for.  I agree it will be helpful to also show which file has the issue.  Let's keep the `file` field. 👍